cobalt/user_agent_platform_info.cc

```diff
@@ -0,0 +1,470 @@
+  info.set_chipset_model_number(
+      base::starboard::SbSysInfo::ChipsetModelNumber());
+  info.set_model_year(base::starboard::SbSysInfo::ModelYear());
+  info.set_brand(base::starboard::SbSysInfo::Brand());
+
+  // Below UA info fields can NOT be retrieved directly from platform's native
+  // system properties.
+
+  // TODO(cobalt, b/374213479): How to retrieve aux_field?
+  info.set_aux_field("dev.cobalt.coat/DeveloperBuild");
+
+  // TODO(cobalt, b/374213479): How to retrieve javascript_engine_version?
+  // C25 retrieves it from cobalt v8c.
+  info.set_javascript_engine_version("v8/8.8.278.8-jit");
+
+  // TODO(cobalt, b/374213479): How to retrieve rasterizer_type_setting?
+  info.set_rasterizer_type("gles");
+
+  // TODO(cobalt, b/374213479): Retrieve Evergreen info.
+  // #if BUILDFLAG(IS_EVERGREEN)
+  //   updater::EvergreenLibraryMetadata evergreen_library_metadata =
+  //       updater::GetCurrentEvergreenLibraryMetadata();
+  //   info.set_evergreen_version(evergreen_library_metadata.version);
+  //   info.set_evergreen_file_type(evergreen_library_metadata.file_type);
+  //   info.set_evergreen_type("Lite");
+  // #endif
+
+  // Retrieve additional platform info.
+
+  // TODO(cobalt, b/374213479): How to retrieve the additional platform info?
+  // C25 builds it using a Cobalt extension.
+  info.set_android_build_fingerprint(
+      "google/sabrina/sabrina:12/STTL.241013.003/12571383:userdebug/dev-keys");
+  info.set_android_os_experience("Amati");
+  info.set_android_play_services_version("244337110");
+
+  // TODO(cobalt, b/374213479): How to determine below Cobalt build macros.
+  // Migrate cobalt/build/build_info.py from C25?
+  info.set_starboard_version("Starboard/17");  // SB_API_VERSION macro in C25.
+  info.set_cobalt_version("26.lts.0");         // COBALT_VERSION in C25.
+  info.set_cobalt_build_version_number(
+      "1234567");                      // COBALT_BUILD_VERSION_NUMBER in C25.
+  info.set_build_configuration("qa");  // COBALT_BUILD_TYPE_DEBUG in C25.
+
+// Apply overrides from command line
+#if !defined(COBALT_BUILD_TYPE_GOLD)
+  if (base::CommandLine::InitializedForCurrentProcess()) {
+    base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
+    const char kUserAgentClientHints[] = "user_agent_client_hints";
+    if (command_line->HasSwitch(kUserAgentClientHints)) {
+      LOG(INFO) << "Entering UA overrides";
+      std::string user_agent_input =
+          command_line->GetSwitchValueASCII(kUserAgentClientHints);
+
+      std::map<std::string, std::string> user_agent_input_map;
+      GetUserAgentInputMap(user_agent_input, user_agent_input_map);
+
+      for (const auto& input : user_agent_input_map) {
+        LOG(INFO) << "Overriding " << input.first << " to " << input.second;
+
+        if (!input.first.compare("starboard_version")) {
+          info.set_starboard_version(input.second);
+          LOG(INFO) << "Set starboard version to " << input.second;
```

Comment:
Here and below, instead of "Set", which can be misinterpreted as a command, consider "Setting", which is not ambiguous.